Assets/_Functionalities/ObjectInformation/Scripts/ObjectSelectorService.cs

```diff
@@ -60,13 +64,55 @@ private void Awake()
             featureSelector = gameObject.AddComponent<FeatureSelector>();
             featureSelector.SetMappingTree(MappingTree);
 
-            foreach(Tool tool  in activeForTools) 
-                tool.onClose.AddListener(() => Deselect());
-
+            ProjectData.Current.OnDataChanged.AddListener(OnProjectChanged);
```

Comment:
Ik denk dat het beter is om dit in OnEnable/OnDisable te doen omdat je nu een ander object accest in Awake